src/coreclr/jit/lclvars.cpp

```diff
@@ -326,6 +326,19 @@ void Compiler::lvaInitTypeRef()
             }
         }
 
+        if ((corInfoTypeWithMod & CORINFO_TYPE_MOD_COPY_WITH_HELPER) != 0)
+        {
+            if (corInfoType == CORINFO_TYPE_VALUECLASS || corInfoType == CORINFO_TYPE_BYREF)
+            {
+                JITDUMP("Setting lvRequiresSpecialCopy for V%02u\n", varNum);
+                varDsc->lvRequiresSpecialCopy = 1;
+            }
+            else
+            {
+                JITDUMP("Ignoring special copy for non-struct, non-byref V%02u\n", varNum);
+            }
+        }
+
```

Comment:
Currently you can see it on a local because the IL stub uses a pinned byref local. I'll see if I can change the IL stub strategy to just use the local for pinning and use the arg for passing.